src/openrct2/ride/TrackStyle.h

```diff
@@ -0,0 +1,103 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "TrackPaint.h"
+
+#include <cstdint>
+
+enum class TrackStyle : uint8_t
+{
+    _3DCinema,
+    airPoweredVerticalCoaster,
+    alpineCoaster,
+    boatHire,
+    bobsleighCoaster,
+    carRide,
+    chairlift,
+    circus,
+    classicStandUpRollerCoaster,
+    classicWoodenRollerCoaster,
+    classicWoodenTwisterRollerCoaster,
+    compactInvertedCoaster,
+    corkscrewRollerCoaster,
+    crookedHouse,
+    dinghySlide,
+    dinghySlideCovered,
+    dodgems,
+    enterprise,
+    facility,
+    ferrisWheel,
+    flyingRollerCoaster,
+    flyingRollerCoasterInverted,
+    flyingSaucers,
+    ghostTrain,
+    goKarts,
+    hauntedHouse,
+    heartlineTwisterCoaster,
+    hybridCoaster,
+    invertedHairpinCoaster,
+    invertedImpulseCoaster,
+    invertedRollerCoaster,
+    juniorRollerCoaster,
+    latticeTriangle,
+    latticeTriangleAlt,
+    launchedFreefall,
+    layDownRollerCoasterInverted,
+    lift,
+    limLaunchedRollerCoaster,
+    logFlume,
+    loopingRollerCoaster,
+    magicCarpet,
+    maze,
+    merryGoRound,
+    mineRide,
+    mineTrainCoaster,
+    miniGolf,
+    miniHelicopters,
+    miniRollerCoaster,
+    miniSuspendedCoaster,
+    miniatureRailway,
+    monorail,
+    monorailCycles,
+    motionSimulator,
+    multiDimensionRollerCoaster,
+    multiDimensionRollerCoasterInverted,
+    observationTower,
+    reverseFreefallCoaster,
+    reverserRollerCoaster,
+    riverRapids,
+    rotoDrop,
+    shop,
+    sideFrictionRollerCoaster,
+    singleRailRollerCoaster,
+    spaceRings,
+    spiralSlide,
+    splashBoats,
+    standUpRollerCoaster,
+    steelWildMouse,
+    steeplechase,
+    submarineRide,
+    suspendedMonorail,
+    suspendedSwingingCoaster,
+    swingingInverterShip,
+    swingingShip,
+    topSpin,
+    twist,
+    twisterRollerCoaster,
+    virginiaReel,
+    waterCoaster,
+    woodenRollerCoaster,
+    woodenWildMouse,
+
```

Comment:
When we add new track styles in the future, will we still be able to keep this list alphabetical?

Reply:
At some point (hopefully sooner rather than later), this number is directly saved in the tile element. Until that point, we can still change this list.

The reason I went with an alphabetical order is mostly because I had to pick something, as this list does not completely correspond with the RIDE_TYPE_* list.